File: ruby/raindrops/raindrops.rb
=begin
Write your code for the 'Raindrops' exercise in this file. Make the tests in
`raindrops_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/raindrops` directory.
=end

class Raindrops
  HASH = {
    3 => 'Pling',
    5 => 'Plang',
    7 => 'Plong'
  }

  def self.convert(n)
    s = ''
    [3,5,7].each { |i| s += HASH[i] if n % i == 0}
    s == '' ? n.to_s : s
  end
end

File: go/raindrops/raindrops.go
package raindrops

import "strconv"

var translations = []struct {
	Num  int
	Word string
}{
	{3, "Pling"},
	{5, "Plang"},
	{7, "Plong"},
}

// Convert func
func Convert(num int) string {
	s := ""
	for _, t := range translations {
		if num%t.Num == 0 {
			s += t.Word
		}
	}
	if s == "" {
		return strconv.Itoa(num)
	}
	return s
}
